### ISSUE TITLE
tf.image.adjust_jpeg_quality/tf.rawOpsEncodeJpegVariableQuality returns negative/inverted RGB image

### ISSUE THREAD

**Issue by fperezgamonal** [tensorflow/tensorflow#40382]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): **No** (only called low-level operators, like `EncodeJpegVariableQuality` from `tf.rawOps`)
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): **Google Colaboratory => Ubuntu 18.04**
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: **N/A**
- TensorFlow installed from (source or binary): **1.X version available in Google Colaboratory**
- TensorFlow version (use command below): **1.15.2**
- Python version: **3.6.9**
- Bazel version (if compiling from source): **N/A**
- GCC/Compiler version (if compiling from source): **7.5.0**
- CUDA/cuDNN version: **10.1 /10.1.243**
- GPU model and memory: **Tesla K80 (11441MiB available)**

**Describe the current behavior**
`tf.image.random_jpeg_quality` (using `tf.image.adjust_jpeg_quality`) returns an image with wrong RGB values (negative image of the one expected)

**Describe the expected behavior**
`tf.image.random_jpeg_quality` samples a different quality at runtime and returns properly formatted RGB image.

**Standalone code to reproduce the issue**
 [Colab Notebook](https://colab.research.google.com/drive/1H1vhKRDwy2E6qLX6XmCtbqW6iv6DiMDj?usp=sharing)

In the standalone code one can see the results of using the following 3 scenarios to emulate jpeg compression given an input **float32** image:
1. Directly call `tf.image.random_jpeg_quality` with two ints (max and min quality).
2. Using custom code to call `tf.image.adjust_jpeg_quality` with tensors not int(s)
3. Construct custom function based on `tf.image.adjust_jpeg_quality` but forcing the call to "EncodeJpegVariableQuality" operator (to ensure a different quality is obtained each run)

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Important notes**
I'm trying to use `tf.image.random_jpeg_quality` in my pipiline to emulate JPEG compression artifacts as another data augmentation transformation. This means that the input to the function is a float32 tensor which internally is converted to uint8, encoded, decoded again and converted back to float32.

However, `tf.image.convert_image_dtype` (in `tf.image.adjust_jpeg_quality`) re-scales the float32 to the (0.0, 1.0) range, reason why I manually re-scale it to (0.0, 255.0) as there are other data augmentations down the line that need standard RGB range (and mean-max normalisation is applied at the end of the pipeline). 
In scenario 3, I've also played around with the argument `saturate` of `tf.image.convert_image_dtype` hoping the problem was caused by an overflow while converting from float32 to uint8, but without success.

The "normal" and "negative" images (manually inverting them after seeing the RGB images outputted after re-scaling the output of the jpeg_quality functions).
The outputs in each case look as follows:
**'Normal' RGB images (without inverting their values)**
![image](https://user-images.githubusercontent.com/19503950/84380349-7d86c200-abe7-11ea-8bac-d0e6c09279f2.png)
(original left,  scenario 1)
![image](https://user-images.githubusercontent.com/19503950/84380718-203f4080-abe8-11ea-9dda-4f728adab232.png)
(Scenario 2, scenario 3)

**'Inverted/Negative' RGB images (after inverting their values)**
![image](https://user-images.githubusercontent.com/19503950/84381052-b4a9a300-abe8-11ea-96df-1d54764b1c91.png)
(original left,  scenario 1)
![image](https://user-images.githubusercontent.com/19503950/84380978-93e14d80-abe8-11ea-9000-2af1e1e09800.png)
(Scenario 2, scenario 3)

As you can see, inverting the RGB values gives a much more plausible image. However, after printing the normal + negative image(s) mean, I see that the average gray value is quite low at around 50 (after inversion). This is more likely the reason why the images look "washed-out".

Moreover, looking at the mean of the float32 image before inputting it to the JPEG functions and after re-scaling and clipping to range, differences are huge:
Input image has a mean (over 3 RGB channels) of ~100, output from jpeg functions is ~205.
Next, I'll try to convert the float32 to uint8 before calling the functions and let you known if anything changes.

Possibly related issues: #25882 (closed)

By all means, this could be an error on my end due to a bad usage of any of the aforementioned functions. I'm only reporting it here so no one else repeats such error or, we resolve the bug, if any.

Thanks in advance!

Regards,
Ferran.

**Comment by fperezgamonal** [comment]
A quick update for anyone experiencing similar results. I've only tested on this new [colab notebook](https://colab.research.google.com/drive/19q3Zec0Mz-2uKqMGpLR74Nqmz5GA7n1d?usp=sharing) but passing a _uint8_ image to any of the JPEG functions and then converting to float32 seems to work just fine and the sampled `quality` is different everytime.

I'll update this reply once I can pass a float32, convert to uint8, encode jpeg, decode and convert again to float32 and obtain the same results.

**Comment by amahendrakar** [comment]
@fperezgamonal,
Thank you for the update. Is this still an issue? Please feel free to close the issue if resolved. Thanks!

**Comment by fperezgamonal** [comment]
Tomorrow morning I'll be able to check it (an older training is underway now) and I'll immediately close this issue if I don't find any other problem.

**Comment by fperezgamonal** [comment]
Hello,

I've just verified that it works as expected with my code, producing different quality images each run given a maximum and minimum values.

Just to summarise, the weird results **were caused by passing a `float32` image outside the usual `float32` range of (0.0, 1.0),** in (0.0, 255.0). As a consequence, `tf.image.convert_image_dtype`  yield an image with wrong RGB values.

I solved this by casting the `float32` in the range (0.0, 255.0) to `uint8`, pass through jpeg functions and cast back to `float32` (same range as original). 
If one wants to use `float32` with the aforementioned jpeg functions, simply normalise your floats in (0.0, 1.0) and then scale back to the original range.
